gen/src/generate/mod.rs

```diff
@@ -0,0 +1,38 @@
+mod char_property;
+
+mod idna;
+mod ucd;
+
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+
+/// The standard auto-generated source disclaimer.
+const PREAMBLE: &'static str = "// WARNING: Auto-generated by unic-gen. DO NOT EDIT MANUALLY!";
+
+pub fn generate_all(mut crates: Vec<String>) -> io::Result<()> {
+    crates.sort();
+
+    for krate in &crates {
+        let path = tables_path(krate);
+        if path.exists() {
+            fs::remove_dir_all(&path)?;
+        }
+        fs::create_dir_all(path)?;
+    }
+
+    idna::generate_all(crates.iter().filter(|x| x.starts_with("unic-idna")))?;
```

Comment:
Now that we're on it, would be great to accept the source (like `ucd` and `idna`) as a CLI argument, optionally, with default being all sources.

This would be again similar to the config matter: there could be cases that we want to download and generate files source-by-source, and having one command to download/generate all would need manual work to revert changes in git before commit.

What do you think?